xline/src/server/watch_server.rs

```diff
@@ -198,17 +198,6 @@ where
 
     /// Handle `WatchCreateRequest`
     async fn handle_watch_create(&mut self, req: WatchCreateRequest) {
-        let compacted_revision = self.kv_watcher.compacted_revision();
-        if req.start_revision < compacted_revision {
-            let result = Err(tonic::Status::invalid_argument(format!(
-                "required revision {} has been compacted, compacted revision is {}",
-                req.start_revision, compacted_revision
-            )));
-            if self.response_tx.send(result).await.is_err() {
-                self.stop_notify.notify(1);
-            }
-            return;
-        }
```

Comment:
Why do we need to remove these?

Reply:
For compatibility with etcd. When using etcdctl to watch a key that has already been compacted, the creation of the watcher can still succeed. However, this watcher will fail during the synchronization process and return the corresponding WatchResponse.